lib/HLSL/DxilShaderModel.cpp

```diff
@@ -28,6 +28,17 @@ ShaderModel::ShaderModel(Kind Kind, unsigned Major, unsigned Minor, const char *
 , m_NumUAVRegs(NumUAVRegs) {
 }
 
+bool ShaderModel::operator==(const ShaderModel &other) const {
+    return m_Kind          == other.m_Kind
+        && m_Major         == other.m_Major
+        && m_Minor         == other.m_Minor
+        && m_pszName       == other.m_pszName
```

Comment:
I thought that the pointer compare would be sufficent because the shader models are constructed from a table with constant string names. A string compare would be more robust though, so I will change it.